lib/notifier/utils.js

```diff
@@ -0,0 +1,195 @@
+const _ = require('lodash/fp')
+const crypto = require('crypto')
+const numeral = require('numeral')
+const prettyMs = require('pretty-ms')
+
+const coinUtils = require('../coin-utils')
+const {
+  CODES_DISPLAY,
+  NETWORK_DOWN_TIME,
+  PING,
+  ALERT_SEND_INTERVAL
+} = require('./codes')
+
+const DETAIL_TEMPLATE = {
+  deviceId: '',
+  cryptoCode: '',
+  code: '',
+  cassette: '',
+  age: '',
+  customerId: '',
+  cryptoAddress: '',
+  direction: '',
+  fiat: '',
+  fiatCode: ''
+}
+
+function parseEventNote(event) {
+  return _.set('note', JSON.parse(event.note), event)
+}
+
+function checkPing(device) {
+  const age = +Date.now() - +new Date(device.lastPing)
+  if (age > NETWORK_DOWN_TIME)
+    return [{ code: PING, age, machineName: device.name }]
+  return []
+}
+
+const getDeviceTime = _.flow(_.get('device_time'), Date.parse)
+
+const isActive = it => it.active && (it.balance || it.errors)
+
+const codeDisplay = code => CODES_DISPLAY[code]
+
+const alertFingerprint = {
+  fingerprint: null,
+  lastAlertTime: null
+}
+
+const getAlertFingerprint = () => alertFingerprint.fingerprint
+
+const getLastAlertTime = () => alertFingerprint.lastAlertTime
+
+const setAlertFingerprint = (fp, time = Date.now()) => {
+  alertFingerprint.fingerprint = fp
+  alertFingerprint.lastAlertTime = time
+}
+
+const shouldNotAlert = currentAlertFingerprint => {
+  return (
+    currentAlertFingerprint === getAlertFingerprint() &&
+    getLastAlertTime() - Date.now() < ALERT_SEND_INTERVAL
+  )
+}
+
+function getAlertTypes(alertRec, config) {
+  let alerts = []
+  if (!isActive(config)) return alerts
+
+  if (config.balance) {
+    alerts = _.concat(alerts, alertRec.general)
+  }
+
+  _.keys(alertRec.devices).forEach(function (device) {
+    if (config.balance) {
+      alerts = _.concat(alerts, alertRec.devices[device].balanceAlerts)
+    }
+
+    if (config.errors) {
+      alerts = _.concat(alerts, alertRec.devices[device].deviceAlerts)
+    }
+  })
```

Comment:
Once again here, logic being repeated
```suggestion
  _.forEach(_.keys(alertRec.devices), (device) => {	
    if (config.balance)
      alerts = _.concat(alerts, alertRec.devices[device].balanceAlerts)

    if (config.errors)
      alerts = _.concat(alerts, alertRec.devices[device].deviceAlerts)
  })
```